test/typescript-tests/nodeExtensions.ts

```diff
@@ -0,0 +1,52 @@
+import { expectTypeOf } from 'expect-type'
+import {
+  MathNode,
+  BaseNode,
+  MathNodeCommon,
+  ConstantNode,
+  Node,
+  FunctionAssignmentNode,
+} from 'mathjs'
+
+interface MyNode extends BaseNode {
+  type: 'MyNode'
+  a: MathNode
+}
+declare module 'mathjs' {
+  interface MathNodeTypes {
+    MyNode: MyNode
+  }
+}
```

Comment:
I didn't see this declaration in your branch...We would not expect some of the tests below to pass without it, right?

TS told me that ambient declarations need to be at the top level of nesting within a file, and I didn't want this to affect the other tests, so I moved this test to a separate file.

Reply:
> I didn't see this declaration in your branch...We would not expect some of the tests below to pass without it, right?

Yep, think you're right.

>TS told me that ambient declarations need to be at the top level of nesting within a file, and I didn't want this to affect the other tests, so I moved this test to a separate file.

Yeah separate file seems fair. We are aiming to start splitting this up anyways.